plugin-server/src/worker/ingestion/action-webhook-formatter.ts

```diff
@@ -0,0 +1,226 @@
+import { format } from 'util'
+
+import { Action, PostIngestionEvent, Team } from '../../types'
+import { getPropertyValueByPath, stringify } from '../../utils/utils'
+
+enum WebhookType {
+    Slack = 'slack',
+    Other = 'other',
+}
+
+// Sync with .../api/person.py and .../lib/constants.tsx
+const PERSON_DEFAULT_DISPLAY_NAME_PROPERTIES = ['email', 'Email', 'name', 'Name', 'username', 'Username', 'UserName']
+
+const TOKENS_REGEX_BRACKETS_EXCLUDED = /(?<=(?<!\\)\[)(.*?)(?=(?<!\\)\])/g
+const TOKENS_REGEX_BRACKETS_INCLUDED = /(?<!\\)\[(.*?)(?<!\\)\]/g
+
+const determineWebhookType = (url: string): WebhookType => {
+    url = url.toLowerCase()
+    if (url.includes('slack.com')) {
+        return WebhookType.Slack
+    }
+    return WebhookType.Other
+}
+
+export class ActionWebhookFormatter {
```

Comment:
For the reviewer: This is almost identical to the old code but with a class approach instead of a functional one (there was so much state being passed around that it was a mess before).

Some things are removed / simplified as I realised they weren't used but mostly it is the same as before

Reply:
> there was so much state being passed around that it was a mess before

Ahhh plugin-server :)